rust target x86_64-pc-solaris doesn't work

```
/builds/rustc-1.37.0/bin/rustc --target=x86_64-pc-solaris test.rs
ulx-0 17:17 /builds/psumbera/FIREFOX-2/TMP: /builds/rustc-1.37.0/bin/rustc --target=x86_64-pc-solaris test.rs
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-pc-solaris` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
```

This was caused by:
https://github.com/rust-lang/rust/issues/40531

And originally reported there too:
https://github.com/rust-lang/rust/issues/40531#issuecomment-538445711

I have tried add symbolic link as follows:

```
$ ls -l /usr/lib/rustlib/
total 11
-r--r--r--   1 root     bin           50 Aug 28 02:47 components
drwxr-xr-x   2 root     bin            6 Aug 28 02:47 etc
-r--r--r--   1 root     bin            2 Aug 28 02:47 rust-installer-version
lrwxrwxrwx   1 root     root          18 Sep  9 15:57 x86_64-pc-solaris -> x86_64-sun-solaris
drwxr-xr-x   3 root     bin            4 Sep  9 15:57 x86_64-sun-solaris
```

But it doesn't help much:
```

$ rustc --target=x86_64-pc-solaris test.rs
error[E0461]: couldn't find crate `std` with expected target triple x86_64-pc-solaris
  |
  = note: the following crate versions were found:
          crate `std`, target triple x86_64-sun-solaris: /usr/lib/rustlib/x86_64-sun-solaris/lib/libstd-560fc60f1e090eba.rlib
          crate `std`, target triple x86_64-sun-solaris: /usr/lib/rustlib/x86_64-sun-solaris/lib/libstd-560fc60f1e090eba.so

error: aborting due to previous error
```

It seems that libstd-560fc60f1e090eba.rlib expects just x86_64-sun-solaris.

@varkor any idea?

--- a/compiler/rustc_session/src/config.rs
+++ b/compiler/rustc_session/src/config.rs
@@ -1433,7 +1433,7 @@ fn parse_target_triple(matches: &getopts::Matches, error_format: ErrorOutputType
                 early_error(error_format, &format!("target file {:?} does not exist", path))
             })
         }
-        Some(target) => TargetTriple::TargetTriple(target),
+        Some(target) => TargetTriple::from_alias(target),
         _ => TargetTriple::from_triple(host_triple()),
     }
 }

--- a/compiler/rustc_target/src/spec/mod.rs
+++ b/compiler/rustc_target/src/spec/mod.rs
@@ -1800,6 +1800,24 @@ impl TargetTriple {
         Ok(TargetTriple::TargetPath(canonicalized_path))
     }
 
+    /// Creates a target triple from its alias
+    pub fn from_alias(triple: String) -> Self {
+        macro_rules! target_aliases {
+            ( $(($alias:literal, $target:literal ),)+ ) => {
+                match triple.as_str() {
+                    $( $alias => TargetTriple::from_triple($target), )+
+                    _ => TargetTriple::TargetTriple(triple),
+                }
+            }
+        }
+
+        target_aliases! {
+            // `x86_64-pc-solaris` is an alias for `x86_64_sun_solaris` for backwards compatibility reasons.
+            // (See <https://github.com/rust-lang/rust/issues/40531>.)
+            ("x86_64-pc-solaris", "x86_64-sun-solaris"),
+        }
+    }
+
     /// Returns a string triple for this target.
     ///
     /// If this target is a path, the file name (without extension) is returned.

